src/tallies/tally_scoring.cpp

```diff
@@ -0,0 +1,2307 @@
+  auto original_bin = dg_match.bins_[i_bin];
+  dg_match.bins_[i_bin] = d_bin;
+
+  // Determine the filter scoring index
+  auto filter_index = 1;
+  for (auto i = 0; i < tally.filters().size(); ++i) {
+    auto i_filt = tally.filters(i);
+    auto& match {simulation::filter_matches[i_filt]};
+    auto i_bin = match.i_bin_;
+    //TODO: off-by-one
+    filter_index += (match.bins_[i_bin] - 1) * tally.strides(i);
+  }
+
+  // Update the tally result
+  auto results = tally_results(i_tally);
+  //TODO: off-by-one
+  #pragma omp atomic
+  results(filter_index-1, score_index, RESULT_VALUE) += score;
+
+  // Reset the original delayed group bin
+  dg_match.bins_[i_bin] = original_bin;
+}
+
+//! Helper function for nu-fission tallies with energyout filters.
+//
+//! In this case, we may need to score to multiple bins if there were multiple
+//! neutrons produced with different energies.
+
+void
+score_fission_eout(Particle* p, int i_tally, int i_score, int score_bin)
+{
+  const Tally& tally {*model::tallies[i_tally]};
+  auto results = tally_results(i_tally);
+  auto i_eout_filt = tally.filters()[tally.energyout_filter_];
+  auto i_bin = simulation::filter_matches[i_eout_filt].i_bin_;
+  auto bin_energyout = simulation::filter_matches[i_eout_filt].bins_[i_bin];
+
+  const EnergyoutFilter& eo_filt
+    {*dynamic_cast<EnergyoutFilter*>(model::tally_filters[i_eout_filt].get())};
+
+  // Note that the score below is weighted by keff. Since the creation of
+  // fission sites is weighted such that it is expected to create n_particles
+  // sites, we need to multiply the score by keff to get the true nu-fission
+  // rate. Otherwise, the sum of all nu-fission rates would be ~1.0.
+
+  // loop over number of particles banked
+  for (auto i = 0; i < p->n_bank; ++i) {
+    auto i_bank = simulation::n_bank - p->n_bank + i;
+    const auto& bank = simulation::fission_bank[i_bank];
+
+    // get the delayed group
+    auto g = bank.delayed_group;
+
+    // determine score based on bank site weight and keff
+    double score = simulation::keff * bank.wgt;
+
+    // Add derivative information for differential tallies.  Note that the
+    // i_nuclide and atom_density arguments do not matter since this is an
+    // analog estimator.
+    if (tally.deriv_ != C_NONE)
+      apply_derivative_to_score(p, i_tally, 0, 0., SCORE_NU_FISSION, &score);
+
+    if (!settings::run_CE && eo_filt.matches_transport_groups_) {
+
+      // determine outgoing energy group from fission bank
+      auto g_out = static_cast<int>(bank.E);
+
+      // modify the value so that g_out = 1 corresponds to the highest energy
+      // bin
+      g_out = eo_filt.n_bins_ - g_out + 1;
+
+      // change outgoing energy bin
+      simulation::filter_matches[i_eout_filt].bins_[i_bin] = g_out;
+
+    } else {
+
+      double E_out;
+      if (settings::run_CE) {
+        E_out = bank.E;
+      } else {
+        E_out = data::energy_bin_avg[static_cast<int>(bank.E)];
+      }
```

Comment:
I feel like the ternary isn't as readable as here, especially since it's split across two lines.  Do you feel strongly that you prefer ternary for initialization?

Reply:
Nah, what you have is ok.